src/main/scala/sbtcrossproject/CrossProject.scala

```diff
@@ -1,184 +1,115 @@
-
-  private def copy(projects: Map[CrossPlatform, Project] = projects): CrossProject =
-    new CrossProject(crossType, projects)
 }
 
 object CrossProject extends CrossProjectExtra {
-  def apply(id: String, platforms: Seq[CrossPlatform], base: File,
-      crossType: CrossType): CrossProject =
-    apply(platforms.map(p => (p, id + p.name)).toMap, base, crossType)
-
-  def apply(ids: Map[CrossPlatform, String], base: File,
-      crossType: CrossType): CrossProject = {
-
-    val sss = sharedSrcSettings(crossType)
-
-    val projects = ids.map { case (p, id) =>
-      val project = Project(id, crossType.dir(p, base)).
-        settings(sss: _*).
-        enablePlugins(p.plugin.toSeq: _*)
-
-      p -> project
-    }
-
-    new CrossProject(crossType, projects)
-  }
 
-  private def sharedSrcSettings(crossType: CrossType) = Seq(
-      unmanagedSourceDirectories in Compile ++= {
-        makeCrossSources(crossType.sharedSrcDir(baseDirectory.value, "main"),
-            scalaBinaryVersion.value, crossPaths.value)
-      },
-      unmanagedSourceDirectories in Test ++= {
-        makeCrossSources(crossType.sharedSrcDir(baseDirectory.value, "test"),
-            scalaBinaryVersion.value, crossPaths.value)
-      }
-  )
-
-  // Inspired by sbt's Defaults.makeCrossSources
-  private def makeCrossSources(sharedSrcDir: Option[File],
-      scalaBinaryVersion: String, cross: Boolean): Seq[File] = {
-    sharedSrcDir.fold[Seq[File]] {
-      Seq.empty
-    } { srcDir =>
-      if (cross)
-        Seq(srcDir.getParentFile / s"${srcDir.name}-$scalaBinaryVersion", srcDir)
-      else
-        Seq(srcDir)
+  def crossProject_impl(c: Context): c.Expr[Builder] = {
+    import c.universe._
+    val enclosingValName = MacroUtils.definingValName(
+        c,
+        methodName =>
+          s"""$methodName must be directly assigned to a val, such as `val x = $methodName`.""")
+    val name = c.Expr[String](Literal(Constant(enclosingValName)))
+    reify {
+      new Builder(name.splice)
     }
   }
 
-  final class Builder(id: String, platforms: Seq[CrossPlatform], base: File) {
-    def crossType(crossType: CrossType): CrossProject =
-      CrossProject(id, platforms, base, crossType)
-  }
-
-  def crossProject_impl(c: Context)(platforms: c.Expr[CrossPlatform]*): c.Expr[Builder] = {
-    import c.universe._
-    val enclosingValName = MacroUtils.definingValName(c, methodName =>
-      s"""$methodName must be directly assigned to a val, such as `val x = $methodName`.""")
-    val name = c.Expr[String](Literal(Constant(enclosingValName)))
-    val SeqApplyPath = Seq("scala", "collection", "Seq", "apply")
-    val SeqApply = SeqApplyPath.foldLeft[Tree](Ident("_root_")) {
-      case (t, name) => Select(t, newTermName(name))
+  final class Builder(id: String) {
+    def in(dir: File): CrossProject[Shared.type] = {
+      new CrossProject(
+          id,
+          dir,
+          Map(Shared -> Project(id, new File(s"$dir/${Shared.path}"))))
```

Comment:
There should be no such thing as a shared `Project`, and therefore a `Shared` `Platform` does not make sense. All there is is a set of shared source directories, but that does not make a shared project.